corehq/apps/app_manager/models.py

```diff
@@ -4444,23 +4475,32 @@ def generate_shortened_url(self, url_type):
         else:
             return shortened_url
 
-    def get_short_url(self):
-        if not self.short_url:
-            self.short_url = self.generate_shortened_url('download_jad')
-            self.save()
-        return self.short_url
-
-    def get_short_odk_url(self, with_media=False):
-        if with_media:
-            if not self.short_odk_media_url:
-                self.short_odk_media_url = self.generate_shortened_url('download_odk_media_profile')
+    def get_short_url(self, build_profile_id=None):
+        if not build_profile_id:
+            if not self.short_url:
+                self.short_url = self.generate_shortened_url('download_jad')
                 self.save()
-            return self.short_odk_media_url
+            return self.short_url
         else:
-            if not self.short_odk_url:
-                self.short_odk_url = self.generate_shortened_url('download_odk_profile')
-                self.save()
-            return self.short_odk_url
+            return self.generate_shortened_url('download_jad', build_profile_id)
```

Comment:
I chose not to save it to the app because it would require one `short_url` for every build profile. `self.short_url` is saved for the default only now where there is no profile specified (99% of use cases) but will be generated each time for the build profiles. I didnt want to overwrite self.short_url each time a new profile was called but also didnt want to create new properties for every possible url